graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -484,6 +495,299 @@ auto GetTemporalPickFn(
 template <typename NumPickFn, typename PickFn>
 c10::intrusive_ptr<FusedSampledSubgraph>
 FusedCSCSamplingGraph::SampleNeighborsImpl(
+    const torch::Tensor& seeds,
+    torch::optional<std::vector<int64_t>>& seed_offsets,
+    const std::vector<int64_t>& fanouts, bool return_eids,
+    NumPickFn num_pick_fn, PickFn pick_fn) const {
+  const int64_t num_seeds = seeds.size(0);
+  const auto indptr_options = indptr_.options();
+
+  // Calculate GrainSize for parallel_for.
+  // Set the default grain size to 64.
+  const int64_t grain_size = 64;
+  torch::Tensor picked_eids;
+  torch::Tensor subgraph_indptr;
+  torch::Tensor subgraph_indices;
+  torch::optional<torch::Tensor> subgraph_type_per_edge = torch::nullopt;
+  torch::optional<torch::Tensor> edge_offsets = torch::nullopt;
+
+  bool with_seed_offsets = seed_offsets.has_value();
+  bool hetero_with_seed_offsets = with_seed_offsets && fanouts.size() > 1;
+
+  // Get the number of edge types. If it's homo or if the size of fanouts is 1
+  // (hetero graph but sampled as a homo graph), set num_etypes as 1.
+  // In temporal sampling, this will not be used for now since the logic hasn't
+  // been adopted for temporal sampling.
+  const int64_t num_etypes =
+      (edge_type_to_id_.has_value() && hetero_with_seed_offsets)
+          ? edge_type_to_id_->size()
+          : 1;
+  std::vector<int64_t> etype_id_to_src_ntype_id(num_etypes);
+  std::vector<int64_t> etype_id_to_dst_ntype_id(num_etypes);
+  torch::Tensor subgraph_indptr_substract;
```

Comment:
Let's make this torch::optional<torch::Tensor>, because it may not have a value.